base/range.jl

```diff
@@ -834,10 +834,10 @@ function getindex(r::AbstractUnitRange, s::AbstractUnitRange{T}) where {T<:Integ
             if first(s)
                 return r
             else
-                return range(r[1], length=0)
+                return range(first(r), length=0)
             end
         else # length(s) == 2
-            return range(r[2], length=1)
+            return range(last(r), length=1)
```

Comment:
I use `last` here and below as we know `length(s) == length(r) == 2` in this branch.